govuk-one-login/authentication-frontend
# AUT-2608: send the correct flags to the .njk error pages

In the reset-password-check-email-controller.ts, depending on the error codes, we redirect to 3 different error page indexes, in the block starting on line #64. Some of the indexes use the show2HrScreen, others the support2hrLockout. Thus, we need to send both of them for all the indexes to work.

## What

Implement redirect to error page when user is locked out due to too many incorrect codes entered.

## How to review


1. Code Review
2. Ensure environment variables 'SUPPORT_2HR_LOCKOUT' and 'SUPPORT_2FA_B4_PASSWORD_RESET' are set to 1
3. Deploy to sandpit with `./deploy-sandpit.sh -l`
4. for the first issue proceed through PW reset journey and enter too many wrong OTP emails and check that the screen you're redirected to after the 5th incorrect attempt is the first screenshot below
5. Restart the journey by going to localhost:2000, or going through the stub again, go to the reset-password journey and verify that you're redirected to the 2nd screenshot below. 


Expected 2hr screens to be redirected to during the PW reset journey, when entering too many wrong codes:

![image](https://github.com/govuk-one-login/authentication-frontend/assets/146068663/72e8f1d4-abad-42d4-acbb-2304d0da3b39)

Expected 2hr screens to be redirected to during the sign-in, PW reset and account recovery journeys, when user is locked and tries to complete the journey: 
![image](https://github.com/govuk-one-login/authentication-frontend/assets/146068663/25f621b4-8ee7-4db1-abb3-1eeea9772192)

Expected 2hr screens to be redirected to during the creation and account part created journeys, when requesting too many codes:
![image](https://github.com/govuk-one-login/authentication-frontend/assets/146068663/02c5812c-2f2f-4225-bd7b-ad0594c54107)

## Change have been demonstrated

Changes to the user interface or content should be demonstrated to Content Design and Interaction Design before being merged. This is to ensure they can make any necessary changes to Figma.

The changes were demonstrated in a demo to Kim Clayden
